table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/AstyanaxTableDAO.java

```diff
@@ -1559,4 +1640,214 @@ protected static String getMillisecondPrecisionZonedDateTime(ZonedDateTime dateT
                 .withZone(ZoneId.of("UTC"))
                 .format(dateTime);
     }
-}
\ No newline at end of file
+
+    @Override
+    public void registerTableListener(String registrationId, Instant newExpirationTime) {
+        Instant now = _clock.instant();
+
+        Delta registrationDelta = TableEventDatacenter.newRegistrant(registrationId, now, newExpirationTime);
+
+        Audit audit = new AuditBuilder()
+                .setComment(String.format("Registering %s", registrationId))
+                .setProgram(registrationId)
+                .setLocalHost()
+                .build();
+
+        _backingStore.update(_systemTableEventRegistry, _selfDataCenter, TimeUUIDs.newUUID(), registrationDelta, audit, WriteConsistency.GLOBAL);
+    }
+
+    @Override
+    public Map.Entry<String, TableEvent> getNextTableEvent(String registrationId) {
```

Comment:
minor: I would say getNextReadyTableEvent to be explicit for the ready events.

Reply:
agreed